lists-service/src/main/java/au/org/ala/listsapi/service/ValidationService.java

```diff
@@ -0,0 +1,137 @@
+package au.org.ala.listsapi.service;
+
+import au.org.ala.listsapi.model.*;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class ValidationService {
+
+  public static final Logger log = LoggerFactory.getLogger(ValidationService.class);
+
+  @Value("${constraints.file}")
+  private String constraintsFile;
+
+  @Value("${userDetails.api.url}")
+  private String userDetailsUrl;
+
+  private Map<String, List<ConstraintListItem>> constraints = null;
+
+  @PostConstruct
+  private void init() throws IOException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    File json = new File(constraintsFile);
+
+    constraints = objectMapper.readValue(json, new TypeReference<HashMap<String, List<ConstraintListItem>>>() {});
+
+    try {
+      List<Location> countries = fetchJson(userDetailsUrl + "ws/registration/countries.json", new TypeReference<>() {});
+
+      // Map the countries list into UI constraints
+      List<ConstraintListItem> countryConstraints = countries.stream().map(e -> {
+        var constraint = new ConstraintListItem();
+        constraint.setLabel(e.getName());
+        constraint.setValue(e.getIsoCode());
+
+        return constraint;
+      }).toList();
+
+      constraints.put("countries", countryConstraints);
```

Comment:
1. Make use of `ConstraintType.countries` instead of the string literal.
2. I think `countryConstraints` should be added to the existing list defined in `constraints.json` (if it exists) instead of overwriting them.
3. I will comment at the top of the review as well, but the region constraint does not exist in the exist lists application. Constraining it in the lists-ui (userdetails countries, as well as userdetails states/territories - for a pre-defined country or countries, and custom entries in constraints.json such as world/none/other/and so on) might need a change from a pick-list to a searchable list, and/or additional metadata for grouping, sorting, section headers, as defined in the requirements document.

Reply:
Fixed in 07271ef6f286a6a56b7fde4ac683bd9f59f5dc9a - changed to use `ConstraintType.countries`, and changed the behaviour so that the countries listed in `constraints.json` are appended to, rather than being overwritten.

I've included `None`, `Other` and `Global` as defaults and `AC` (Ashmore and Cartier Islands) & `CS` (Coral Sea Islands) as they're not included in the userdetails `countries.json` file by default.